### PR TITLE
Enabling standalone builds for Test Project of any Priority

### PR DESCRIPTION
**Pull request by ramarag**
@wtgodbe  @bryanar PTAL

FYI : @rahku @gkhanna79  @janvorli

### CODE REVIEW

**Comment by ramarag** [comment]
@dotnet-bot  test windows_nt arm64

File: tests/src/dir.targets [hunk 1]
```diff
@@ -64,14 +64,22 @@
   <!-- If we are a run-only, that depends on another project, this is the "Build" we use. I.e. build all dependency projects, absolutely.
   -->
 
-  <Target Name="Build" Condition="('$(CLRTestKind)'=='RunOnly') And ('$(CLRTestPriority)' &lt;= '$(CLRTestPriorityToBuild)')">
+  <PropertyGroup> 
+    <BuildAllProjects Condition=" '$(BuildAllProjects)' == ''">false</BuildAllProjects>
+    <_CLRTestBuildPrioritySatisfied Condition=" '$(_CLRTestBuildPrioritySatisfied)' == ''">false</_CLRTestBuildPrioritySatisfied>
+    <_CLRTestBuildPrioritySatisfied Condition="'$(CLRTestPriority)' &lt;= '$(CLRTestPriorityToBuild)'">true</_CLRTestBuildPrioritySatisfied>
+    <_WillCLRTestProjectBuild Condition="Exists('$(ToolsDir)Build.Common.targets') And $(_CLRTestCompilesSource)">true</_WillCLRTestProjectBuild>
```
Comment (rahku) [inline tests/src/dir.targets]: why is this condition required.....Exists('$(ToolsDir)Build.Common.targets') is it valid for this to be false under any circumstance.

File: tests/src/Common/test_runtime/test_runtime.csproj [hunk 2]
```diff
@@ -15,4 +13,5 @@
     <None Include="project.json" />
   </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
-</Project>
\ No newline at end of file
+  <Target Name="Build" />
```
Comment (rahku) [inline tests/src/Common/test_runtime/test_runtime.csproj]: it still generates project.lock.json file I guess

File: tests/src/dir.targets (hunk 1 above)
Comment (ramarag) [inline tests/src/dir.targets]: The Build.Common.targets file is the build logic which we resue from buildtools, so this file needs to be brought in as part of the nuget restore

File: tests/src/Common/test_runtime/test_runtime.csproj (hunk 2 above)
Comment (ramarag) [inline tests/src/Common/test_runtime/test_runtime.csproj]: yes

File: tests/src/dir.targets (hunk 1 above)
Comment (rahku) [inline tests/src/dir.targets]: Isn't it safe to assume that this file should always be present. why is there a special check for its presence? Is there a scenario where this file is absent and that is fine for the build to succeed?

**Comment by ellismg** [comment]
Did you mean to commit `tests/src/.dir.targets.swp`?

**Comment by wtgodbe** [comment]
So how do we tell build.cmd to build all tests of Priority 1? All tests of priority 1 or less?

**Comment by ramarag** [comment]
build.cmd skipmscorlib skipnative Priority 1 should build the tests as is

**Comment by wtgodbe** [comment]
What do you mean as is? Will that build Priority 1 tests only, or Priority 0 and 1? How would I go about building only Pri 1 tests? There are places that depend on "build.cmd Priority 1" building Priority 1 and Priority 0 tests.

**Comment by ramarag** [comment]
the existing behavior should remain in place

**Comment by ramarag** [comment]
@dotnet-bot  test Windows Pri 1

**Comment by wtgodbe** [comment]
@dotnet-bot test Windows_NT Pri 1

File: tests/src/dir.targets (hunk 1 above)
Comment (ramarag) [inline tests/src/dir.targets]: you are right removed it

**Comment by ramarag** [comment]
@dotnet-bot test Windows_NT pri1

**Comment by ramarag** [comment]
@dotnet-bot test Windows_NT pri1 please.

**Comment by ramarag** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test

**Comment by ramarag** [comment]
@dotnet-bot test  Windows_NT x64 Release Priority 1 Build and Test

**Comment by ramarag** [comment]
@dotnet-bot  Ubuntu x64 Checked Build and Test

**Comment by ramarag** [comment]
@dotnet-bot test  Windows_NT x64 Release Priority 1 Build and Test
                   test Ubuntu x64 Checked Build and Test

**Comment by ramarag** [comment]
This change involves test build infrastructure and has nothing to do with the way test runs. So merging it

**Pull request merged**